src/resources/MachineLearning/Models/Models.ts

```diff
@@ -7,8 +7,11 @@ import {MLModel} from './ModelsInterfaces';
 export default class Models extends Resource {
     static baseUrl = `/rest/organizations/${API.orgPlaceholder}/machinelearning/models`;
 
-    list() {
-        return this.api.get<MLModel[]>(Models.baseUrl);
+    list(engines?: string[]) {
+        const requestUrl = Boolean(engines)
+            ? `${Models.baseUrl}?${engines.map((id) => `engines=${id}`).join('&')}`

```

Comment:
The `id` should be URL-encoded here. The engine ID could be used to inject arbitrary query parameters.